### PR TITLE
[MOBILE-1165] - Add a message center screen to the ReactNative sample app

### PR DESCRIPTION
**Pull request by khmMouna**
### What do these changes do?
Add a message center screen to the ReactNative sample app.

### How did you verify these changes?
Using the sample app.

### Verification Screenshots:
messageCenter

### Anything else a reviewer should know?
I tried hard to display the message center embedded in the navigation tab but I did not succeed.
If anyone know how to do it, I want really to improve it.

### CODE REVIEW

File: example/screens/MessageCenterScreen.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  componentDidMount() {
+    this.handleMessageCenterDisplay()
+  }
+
+  handleMessageCenterDisplay() {
+    UrbanAirship.displayMessageCenter()
+  }
+
+
+  render() {
```
Comment (rlepinski) [inline example/screens/MessageCenterScreen.js]: The ticket must not of been clear, you need to show a message center using react native components not just a button to show the native message center.

Should be two screens:
- Listing of messages using React Native ListView
- A message using the UAMessageView

File: example/App.js [hunk 2]
```diff
@@ -6,15 +6,48 @@
...
-const TabNavigator = createBottomTabNavigator(
-  {
-    Home: HomeScreen,
-    Settings: SettingsScreen,
-  },
+const MessageCenterStack = createStackNavigator({
+ MessageList: {
+  screen: MessageListScreen,
+  navigationOptions: {
+    headerTitle: 'Messages list',
```
Comment (BrianBatchelder) [inline example/App.js]:
```suggestion
    headerTitle: 'Messages Center',
```

File: example/App.js [hunk 3]
```diff
@@ -6,15 +6,48 @@
...
+  }
+ },
+ MessageDetails: {
+  screen: MessageDetailsScreen,
+  navigationOptions: {
+    headerTitle: 'Message details',
```
Comment (BrianBatchelder) [inline example/App.js]: I'm not sure how to make this change, but the title of this screen should be the title of the message being displayed.

File: example/App.js [hunk 4]
```diff
@@ -6,15 +6,48 @@
 
 import { createAppContainer } from 'react-navigation';
 import { createBottomTabNavigator } from 'react-navigation-tabs';
+import { createStackNavigator } from 'react-navigation-stack';
 
 import HomeScreen from "./screens/HomeScreen";
 import SettingsScreen from "./screens/SettingsScreen";
+import MessageListScreen from "./screens/MessageCenterScreen";
+import MessageDetailsScreen from "./screens/MessageScreen";
```
Comment (BrianBatchelder) [inline example/App.js]: I suggest making these more consistent, e.g.:
```suggestion
import MessageCenterScreen from "./screens/MessageCenterScreen";
import MessageScreen from "./screens/MessageScreen";
```

File: example/screens/MessageCenterScreen.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  ScrollView,
+  Button,
+  FlatList,
+  TouchableHighlight,
+} from 'react-native';
+
+import Moment from 'moment';
+
+import styles from './../Styles';
+import MessageScreen from "./MessageScreen"
```
Comment (BrianBatchelder) [inline example/screens/MessageCenterScreen.js]: Is this needed? I don't see it used. I'm not a JS expert, so I could be missing it.

File: example/screens/MessageScreen.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  constructor(props) {
+    super (props);
+  }
+
+  render() {
+    const { params } = this.props.navigation.state;
+    const messageId = params ? params.messageId : "";
+    return (
+       <View style={styles.backgroundContainer}>
+            <UAMessageView messageId={messageId} style={{ flex: 1 }}/>
```
Comment (rlepinski) [inline example/screens/MessageScreen.js]: Could you add error handling and a loading indicator?

**Comment by rlepinski** [comment]
Looking really great. I think you need to add a message listener to deep link to a message from a push. Could you verify that works?

File: example/screens/MessageCenterScreen.js (hunk 5 above)
Comment (khmMouna) [reply example/screens/MessageCenterScreen.js]: Yes you're right, we don't need the `import MessageScreen from "./MessageScreen"` I removed it

File: example/App.js (hunk 3 above)
Comment (khmMouna) [reply example/App.js]: Good idea, I fix it 👍

File: example/screens/MessageScreen.js (hunk 6 above)
Comment (khmMouna) [reply example/screens/MessageScreen.js]: Yes 👍

**Pull request merged**